app/components/search_bar/search_bar.android.js

```diff
@@ -3,176 +3,225 @@
         placeholder: 'Search',
         showCancelButton: true,
-        placeholderTextColor: '#bdbdbd',
-        textColor: '#212121',
+        placeholderTextColor: changeOpacity('#000', 0.5),
         onSearchButtonPress: () => true,
         onCancelButtonPress: () => true,
         onChangeText: () => true,
-        onFocus: () => true,
-        onBlur: () => true
+        onFocus: () => true
     };
 
     constructor(props) {
         super(props);
         this.state = {
-            isOnFocus: false,
-            value: this.props.text
+            isFocused: false,
+            value: ''
         };
-        this.onFocus = this.onFocus.bind(this);
-        this.onBlur = this.onBlur.bind(this);
-        this.onCancelButtonPress = this.onCancelButtonPress.bind(this);
-        this.onSearchButtonPress = this.onSearchButtonPress.bind(this);
-        this.onChangeText = this.onChangeText.bind(this);
     }
 
-    onSearchButtonPress() {
-        const onSearchButtonPress = this.props.onSearchButtonPress;
-        if (this.state.value) {
-            onSearchButtonPress(this.state.value);
+    cancel = () => {
+        this.onCancelButtonPress();
+    };
+
+    onSearchButtonPress = () => {
+        const {onSearchButtonPress} = this.props;
+        const {value} = this.state;
+
+        if (value && onSearchButtonPress) {
+            onSearchButtonPress(value);
         }
-    }
+    };
 
-    onCancelButtonPress() {
-        const onCancelButtonPress = this.props.onCancelButtonPress;
-        this.setState({
-            isOnFocus: false,
-            value: ''
-        });
+    onCancelButtonPress = () => {
+        const {onCancelButtonPress} = this.props;
 
-        onCancelButtonPress();
         Keyboard.dismiss();
-    }
+        InteractionManager.runAfterInteractions(() => {
+            this.setState({
+                isFocused: false,
+                value: ''
+            }, () => {
+                if (onCancelButtonPress) {
+                    onCancelButtonPress();
+                }
+            });
+        });
+    };
 
-    onChangeText(value) {
-        const onChangeText = this.props.onChangeText;
+    onChangeText = (value) => {
+        const {onChangeText} = this.props;
         this.setState({value});
-        onChangeText(value);
-    }
+        if (onChangeText) {
+            onChangeText(value);
+        }
+    };
 
-    onFocus() {
-        const onFocus = this.props.onFocus;
-        this.setState({isOnFocus: true});
-        onFocus();
-    }
+    onFocus = () => {
+        const {onFocus} = this.props;
 
-    onBlur() {
-        const onBlur = this.props.onBlur;
-        this.setState({isOnFocus: false});
-        onBlur();
-        Keyboard.dismiss();
-    }
+        this.setState({isFocused: true});
 
-    blur = () => {
-        this.onBlur();
+        if (onFocus) {
+            onFocus();
+        }
     };
 
     render() {
         const {
-            height,
+            autoCapitalize,
```

Comment:
I don't think it's a problem how it is, but I'm looking for a way to save time later since we wouldn't have to add all the prop types and such